### DISCUSSION TITLE
filter cells by a dataframe

### DISCUSSION THREAD

**Discussion by yamihn** [GreenleafLab/ArchR#1542] [Questions / Documentation]
Hello, my intention is to filter the number of cells of my archr project to obtain the exact number of cells obtained by Seurat object.
In my Seurat object analysis I got 29'844 cells, in Archr I have 39'871.
I created a dataframe in which I have the cellnames of Seurat:
c <- data.frame(cell_slim=rownames(a))
head(c$cell_slim)
[1] "KO_d6_r1_AAACAGCCACCTGCTC-1" "KO_d6_r1_AAACAGCCACGTAATT-1" "KO_d6_r1_AAACAGCCATAATGAG-1" "KO_d6_r1_AAACATGCACCTAATG-1"
[5] "KO_d6_r1_AAACATGCAGGAATCG-1" "KO_d6_r1_AAACATGCAGGATAAC-1"

Could you suggest me a way to take in archr only this cell?
I tried 
archr_subset= subsetArchRProject(
  ArchRProj = Archr,
  cells = getCellNames(c$cell_slim),
  outputDirectory = "~/ArchR",
  dropCells = TRUE,
  logFile = NULL,
  threads = getArchRThreads(),
  force = FALSE
)
but I got this error:
Error in .validInput(input = ArchRProj, name = "ArchRProj", valid = "ArchRProject") : 
  Input value for 'ArchRProj' is not a archrproject, (ArchRProj = character) please supply valid input!

Thank you in advance

**Comment by rcorces** [comment]
you are not using `getCellNames()` correctly, which is why you are getting an error. Please make sure to read all function documentation.
https://www.archrproject.com/reference/getCellNames.html
`getCellNames()` only accepts an `ArchRProj` object.

As stated in the function documentation, the `cells` param expects:
"A vector of cells to subset ArchRProject by"
https://www.archrproject.com/reference/subsetArchRProject.html

Please read all documentation carefully before posting to GitHub. For problems that could be easily solved by reading the function documentation, we may choose not to reply.
